Sources/Core/Client/Client+RxWatcherCount.swift

```diff
@@ -13,16 +13,17 @@ import RxSwift
 public extension Reactive where Base == Client {
     /// Observe a watcher count of users for a given channel.
     func watcherCount(channel: Channel) -> Observable<Int> {
-        queryChannel(channel, messagesPagination: .limit(1), options: [.watch, .state])
-            .map { $0.channel }
-            .flatMapLatest({ [unowned base] channel -> Observable<Int> in
-                base.rx.events(for: [.userStartWatching,
-                                     .userStopWatching,
-                                     .messageNew,
-                                     .notificationMessageNew],
-                               cid: channel.cid)
-                    .map { _ in channel.watcherCount }
-                    .startWith(channel.watcherCount)
-            })
+        .create { (observer) -> Disposable in
+            let subscription = channel.subscribeToWatcherCount { (result) in
+                do {
+                    let response = try result.get()
+                    observer.onNext(response)
+                } catch {
+                    observer.onError(error)
+                }
```

Comment:
You do a lot of copy/paste this. Please check how it was implemented to avoid this repetition.

Reply:
Can you clarify? This is not copy/paste, this just relays the logic into LLC. This is not repetition either?

Reply:
I think we can simplify that as it implemented to bind rx to completion block. Check `AutoCancellingSubscription` file.
If you can't, skip it.

Reply:
You mean just return `channel.rx.watcherCount`?

Reply:
No, all implementations should be on the `Client` (extensions) and on the `Channel` side is a short/handy way.

Reply:
`Client.subscribeToWatcherCount(for channel` is LLC internal, so I can't use it here